### PR TITLE
Event coloring - different approach

### PR DESCRIPTION
**Pull request by jeremypw**
Create one styleclass per calendar rather than per event.

Still gets the same color per event but this is due to a deeper problem with event creation not with the coloring logic.

### CODE REVIEW

File: src/Indicator.vala [hunk 1]
```diff
@@ -210,18 +210,14 @@ public class DateTime.Indicator : Wingpanel.Indicator {
...
 
-            /* Color events per calendar
-             * XXX: Day counter of trying this without help: 5 (Remove this when successful)
-             */
-            foreach (E.Source source in e.sources) {
-                var cal = (E.SourceCalendar)source.get_extension (E.SOURCE_EXTENSION_CALENDAR);
+            /* Color menuitem per calendar source of event */
+            var css_class = Util.get_style_calendar_color (e.cal);
+            menuitem.get_style_context ().add_class (css_class);
+            menuitem_icon.get_style_context ().add_class (css_class);
```
Comment (lainsce) [inline src/Indicator.vala]: I think there's no need for this line, because `menuitem_icon` is inside `menuitem` so the coloring is dealt to both.

**Comment by lainsce** [comment]
Left a comment, but this PR solves everything! :rainbow:

**Comment by lainsce** [comment]
All good after testing. Merging~

**Comment by jeremypw** [comment]
There are some critical warnings generated in the terminal - I need to check whether this PR is responsible.

**Pull request merged**